### PR TITLE
Fix where clause for non-matching {downtime,comment}history IDO database updates

### LINKED ISSUE #5458: IDO donwtimehistory records orphaned from scheduleddowntime records following restart
Following a restart of Icinga2, records in the the downtimehistory table may have internal_downtime_id's that no longer match the correct record in the scheduleddowntime table.  As a result, these records do not get updated correctly when the Downtime ends or is cancelled.

## Expected Behavior

The downtimehistory records should get updated correctly (actual_end_time set, is_in_effect updated, was_cancelled, etc) regardless of whether a restart/reload of Icinga2 was performed prior to the Downtime period.

## Current Behavior

The history table may indicate that downtime started, but never ended, despite the fact that the object is no longer in downtime.

## Possible Solution

I've implemented a very rough workaround by running a cron job and checking that the object_id fields in scheduleddowntime, and downtimehistory records match up... and updating the internal_downtime_id field in the history table accordingly when they don't.  It seems to me that the internal_downtime_id values should never get re-used, and simply be carried through a restart of the system.

## Steps to Reproduce (for bugs)

1. Schedule downtime for several objects in the system (more than one).
2. Restart icinga2
3. Cancel the downtimes or let them expire.
4. Check downtime history in IDO database or via a web interface (like Icinga Web 2)... some objects will show downtime starting, but never ending.

## Context

Not having accurate downtime history recorded in the database makes it hard to do availability reporting where downtime states a segregated out for informational purposes.  (problems during planned downtime often aren't supposed to affect overall SLA calculations, for example).

## Your Environment

* Version used (`icinga2 --version`): 2.7.0
* Operating System and version: Centos &
* Enabled features (`icinga2 feature list`):
api checker command ido-mysql livestatus mainlog
* Icinga Web 2 version and modules (System - About):
* Config validation (`icinga2 daemon -C`):
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes.

### PR DESCRIPTION
**Pull request by Crunsher**
This is our partial fix for 2.7.1 full fix with new index will come in 2.8.0. See #5485

fixes #5458 #5492

### CODE REVIEW

File: lib/db_ido/dbevents.cpp [hunk 1]
```diff
@@ -483,7 +482,6 @@ void DbEvents::AddDowntimeInternal(std::vector<DbQuery>& queries, const Downtime
 		return;
 	}
 
-	fields1->Set("name", downtime->GetName());
```
Comment (dnsmichi) [inline lib/db_ido/dbevents.cpp]: Why is this removed?

**Comment by Crunsher** [comment]
With so many tickets it's hard to keep track what is actually requested or meant :/ Should now be correct

**Comment by Crunsher** [comment]
I also messed up the query names, now it should all be fine.

**Comment by dnsmichi** [comment]
Branch was not rebased.

There were some left-overs with where conditions using the maybe-changed internal_*_id for runtime updates after restart, and the yet more important query: trigger downtime.

Amended and tested them.

## Simple test

Scheduled and triggered:

```
*************************** 10. row ***************************
    downtimehistory_id: 10
           instance_id: 1
         downtime_type: 1
             object_id: 403
            entry_time: 2017-09-20 11:57:36
           author_name: icingaadmin
          comment_data: fdsf
  internal_downtime_id: 12
       triggered_by_id: 0
              is_fixed: 1
              duration: 0
  scheduled_start_time: 2017-09-20 11:57:26
    scheduled_end_time: 2017-09-20 11:59:26
           was_started: 1
     actual_start_time: 2017-09-20 11:57:36
actual_start_time_usec: 658941
       actual_end_time: NULL
  actual_end_time_usec: 0
         was_cancelled: 0
          is_in_effect: 1
          trigger_time: 2017-09-20 11:57:36
                  name: many-test-0!many-test-75!mbmif.int.netways.de-1505901456-9
    endpoint_object_id: 214
10 rows in set (0.00 sec)
```

Restart Icinga 2.

Wait until the downtime expires.

Look into the database again.

```
*************************** 10. row ***************************
    downtimehistory_id: 10
           instance_id: 1
         downtime_type: 1
             object_id: 403
            entry_time: 2017-09-20 11:57:36
           author_name: icingaadmin
          comment_data: fdsf
  internal_downtime_id: 12
       triggered_by_id: 0
              is_fixed: 1
              duration: 0
  scheduled_start_time: 2017-09-20 11:57:26
    scheduled_end_time: 2017-09-20 11:59:26
           was_started: 1
     actual_start_time: 2017-09-20 11:57:36
actual_start_time_usec: 658941
       actual_end_time: 2017-09-20 11:59:38
  actual_end_time_usec: 721931
         was_cancelled: 0
          is_in_effect: 0
          trigger_time: 2017-09-20 11:57:36
                  name: many-test-0!many-test-75!mbmif.int.netways.de-1505901456-9
    endpoint_object_id: 214
10 rows in set (0.00 sec)
```

### Simulate an ID change


Schedule a downtime and look into the history.

```
MariaDB [icinga]> select * from icinga_downtimehistory order by entry_time desc limit 1\G
*************************** 1. row ***************************
    downtimehistory_id: 21
           instance_id: 1
         downtime_type: 1
             object_id: 1267
            entry_time: 2017-09-20 12:02:15
           author_name: icingaadmin
          comment_data: fdsfsdfds
  internal_downtime_id: 22
       triggered_by_id: 0
              is_fixed: 1
              duration: 0
  scheduled_start_time: 2017-09-20 12:02:08
    scheduled_end_time: 2017-09-20 12:10:08
           was_started: 1
     actual_start_time: 2017-09-20 12:02:15
actual_start_time_usec: 645349
       actual_end_time: NULL
  actual_end_time_usec: 0
         was_cancelled: 0
          is_in_effect: 1
          trigger_time: 2017-09-20 12:02:15
                  name: many-test-10!many-test-108!mbmif.int.netways.de-1505901735-10
    endpoint_object_id: 214
1 row in set (0.00 sec)
```

Stop Icinga 2 and update the row.

```
MariaDB [icinga]> update icinga_downtimehistory set internal_downtime_id=999 where downtimehistory_id=21;
Query OK, 1 row affected (0.00 sec)
Rows matched: 1  Changed: 1  Warnings: 0

MariaDB [icinga]> select * from icinga_downtimehistory order by entry_time desc limit 1\G
*************************** 1. row ***************************
    downtimehistory_id: 21
           instance_id: 1
         downtime_type: 1
             object_id: 1267
            entry_time: 2017-09-20 12:02:15
           author_name: icingaadmin
          comment_data: fdsfsdfds
  internal_downtime_id: 999
       triggered_by_id: 0
              is_fixed: 1
              duration: 0
  scheduled_start_time: 2017-09-20 12:02:08
    scheduled_end_time: 2017-09-20 12:10:08
           was_started: 1
     actual_start_time: 2017-09-20 12:02:15
actual_start_time_usec: 645349
       actual_end_time: NULL
  actual_end_time_usec: 0
         was_cancelled: 0
          is_in_effect: 1
          trigger_time: 2017-09-20 12:02:15
                  name: many-test-10!many-test-108!mbmif.int.netways.de-1505901735-10
    endpoint_object_id: 214
1 row in set (0.00 sec)
```

The ID is now different to what Icinga 2 knows about it. (REST API)

screen shot 2017-09-20 at 12 05 33

```
MariaDB [icinga]> select * from icinga_downtimehistory order by entry_time desc limit 1\G
*************************** 1. row ***************************
    downtimehistory_id: 21
           instance_id: 1
         downtime_type: 1
             object_id: 1267
            entry_time: 2017-09-20 12:02:15
           author_name: icingaadmin
          comment_data: fdsfsdfds
  internal_downtime_id: 999
       triggered_by_id: 0
              is_fixed: 1
              duration: 0
  scheduled_start_time: 2017-09-20 12:02:08
    scheduled_end_time: 2017-09-20 12:10:08
           was_started: 1
     actual_start_time: 2017-09-20 12:02:15
actual_start_time_usec: 645349
       actual_end_time: 2017-09-20 12:10:50
  actual_end_time_usec: 249106
         was_cancelled: 0
          is_in_effect: 0
          trigger_time: 2017-09-20 12:02:15
                  name: many-test-10!many-test-108!mbmif.int.netways.de-1505901735-10
    endpoint_object_id: 214
1 row in set (0.01 sec)
```


Start Icinga 2 and wait until the downtime expires (12:10).


```
MariaDB [icinga]> select * from icinga_downtimehistory order by entry_time desc limit 1\G
*************************** 1. row ***************************
    downtimehistory_id: 21
           instance_id: 1
         downtime_type: 1
             object_id: 1267
            entry_time: 2017-09-20 12:02:15
           author_name: icingaadmin
          comment_data: fdsfsdfds
  internal_downtime_id: 999
       triggered_by_id: 0
              is_fixed: 1
              duration: 0
  scheduled_start_time: 2017-09-20 12:02:08
    scheduled_end_time: 2017-09-20 12:10:08
           was_started: 1
     actual_start_time: 2017-09-20 12:02:15
actual_start_time_usec: 645349
       actual_end_time: 2017-09-20 12:10:50
  actual_end_time_usec: 249106
         was_cancelled: 0
          is_in_effect: 0
          trigger_time: 2017-09-20 12:02:15
                  name: many-test-10!many-test-108!mbmif.int.netways.de-1505901735-10
    endpoint_object_id: 214
1 row in set (0.01 sec)
```

actual_end_time is set, means the fix really works.

@Crunsher please include these tests next time :)

**Pull request merged**